HW_3/task_2.py

```diff
@@ -0,0 +1,21 @@
+# Задача 18: Требуется найти в массиве A самый близкий по величине элемент к заданному числу X. Если таких элементов несколько, вы вывести один любой.
+# Пользователь в первой строке вводит натуральное число N – количество элементов в массиве. В последующих  строках записаны N целых чисел Ai (можно использовать псевдогенерацию).
+# Последняя строка содержит число X
+# *Пример:*
+# 5
+#     7 -2 3 5 1
+#     6
+#     -> 7
+
+import random
+
+n_for_array = int(input("Ведите число N, которое соответствует числу элементов в массиве чисел: "))
+x_number = int(input("Ведите число X, которое требуется найти среди элементов в массиве чисел N: "))
+
+array_a = [random.randint(0, 5) for i in range(n_for_array)]
+print(f"Полученный массив чисел {array_a}")
+find_number = 0
+
+for i in array_a:
+   find_number = min(array_a, key = lambda i: abs(i - x_number)) 
```

Comment:
Смысла использовать цикла при таком поиске ближайшего нет. Результатом каждой итерации будет одно и тоже число. Такое использование мне говорит о том, что это код не ваш, так как вы не до конца понимаете как он работает. Пожалуйста, используйте только те инструменты, которые вы знаете.
```suggestion
find_number = min(array_a, key = lambda i: abs(i - x_number)) 
```